### PR TITLE
Rework host tools

### PR DESCRIPTION
**Pull request by ariscop**
Now configured at the same time as reactos and only
one list of targets to maintain

Correctly rebuilds when a tool changes (tested widl)

Needs some testing still, arm currently needs a separate host-tools for example

Related JIRA issues: [CORE-10450](https://jira.reactos.org/browse/CORE-10450) [CORE-12093](https://jira.reactos.org/browse/CORE-12093)

### CODE REVIEW

**Comment by learn-more** [comment]
So basically, remove all configurations around new style builds?
Does nothing in our infra use that option anymore?

**Comment by ariscop** [comment]
new style is the default, and the change here behaves similar
the goal is to have a unified build like new style, but rely on export(), which should be more robust than the original implementation

it also makes building host-tools separately trivial: [6311bd1](/reactos/reactos/pull/1356/commits/6311bd15d108d825bfb01adcc98de6d74850c05a)

**Comment by HeisSpiter** [comment]
> So basically, remove all configurations around new style builds?
> Does nothing in our infra use that option anymore?

I don't think to anything on our infrastructure that would be using "old" build style. Builders have been using new build style since its begin thanks to the default case in CMake configuration.

**Comment by ariscop** [comment]
Has anyone got the arm build working? configure.cmd does some stuff with host-tools but it doesn't seem to work, it looks like cmake doesn't capture lib directories causing link errors

**Comment by ariscop** [comment]
This seems plenty robust and i'd call it done
Haven't found a way to break a build with this new setup, only 'lost' feature is automatically building host-tools for arm builds, but that seems broken in master regardless, along with the arm build itself

File: configure.cmd [hunk 1]
```diff
@@ -151,7 +135,6 @@ REM Parse command line parameters
                     set CMAKE_GENERATOR="Visual Studio 11 Win64"
                 ) else if "!ARCH!" == "arm" (
                     set CMAKE_GENERATOR="Visual Studio 11 ARM"
-                    set CMAKE_GENERATOR_HOST="Visual Studio 11"
```
Comment (learn-more) [inline configure.cmd]: Why are these removed?
In some form or way, the host tools should be built with the 'native' toolset, not with the `ARM` toolset

Comment (Extravert-ir) [reply configure.cmd]: The only place where it's used is being removed...
https://github.com/reactos/reactos/pull/1356/files#diff-3ac55282d165a792e56e4f7af827fd90L266

Comment (learn-more) [reply configure.cmd]: so this, in turn should also not be removed?
Or get an alternative code path.

Comment (ariscop) [reply configure.cmd]: It doesn't seem to work as intended, or at least i couldn't get it to work, the replacement is configuring and building host tools separately and using `-DWITH_HOST_TOOLS=path`

**Comment by Mouvedia** [comment]
> Has anyone got the arm build working?

Clearly [you are not alone](https://twitter.com/CdsmatheusSilva/status/1108573334772596736).

File: configure.cmd (hunk 1 above)
Comment (learn-more) [reply configure.cmd]: `This is the only occurrence that I could find of `WITH-HOST-TOOLS`:
https://git.reactos.org/?p=reactos.git;a=blob;f=configure.sh;hb=a976ebb0315a0b27438edb04fd69555e7de732df#l37

Comment (ariscop) [reply configure.cmd]: its a cmake argument, https://github.com/reactos/reactos/blob/6ad548a49153bd28e47b2e6d30543faa6a0b7ef2/sdk/cmake/host-tools.cmake#L48-L58

Comment (learn-more) [reply configure.cmd]: My bad, I checked in master instead of in your pr.

**Comment by HBelusca** [comment]
@ariscop : could you please adjust your PR, following the committed simplifications made for the MSVC generator?

**Comment by ariscop** [comment]
@HBelusca updated

File: sdk/cmake/host-tools.cmake [hunk 2]
```diff
@@ -31,15 +31,30 @@ function(setup_host_tools)
...
+    include(${HOST_TOOLS_DIR}/TargetList.cmake)
 
     foreach(_target ${NATIVE_TARGETS})
         add_dependencies(native-${_target} build-host-tools)
     endforeach()
 
 endfunction()
+
+function(setup_host_tools)
+    if(WITH_HOST_TOOLS)
```
Comment (MakiseKurisu) [inline sdk/cmake/host-tools.cmake]: I do not see any code set this variable. Does this mean that the ARM host tools needs to be built manually first?

Comment (ariscop) [reply sdk/cmake/host-tools.cmake]: On windows with msvc yes, which is done by running cmake directly and without the TOOLCHAIN_FILE argument
with gcc the host tools should build automatically, its why i unset CC and CXX here
https://github.com/reactos/reactos/blob/9370741ce70034f0a0647983d4ec98522ea2fc8e/sdk/cmake/host-tools.cmake#L6-L12

Comment (MakiseKurisu) [reply sdk/cmake/host-tools.cmake]: Then some comments in configure.cmd to explain this process might be helpful since this is an extra step.

**Comment by learn-more** [comment]
@ColinFinck @HeisSpiter any objection for the builders to merge this?

**Comment by andrew-boyarshin** [comment]
@learn-more it would be great to see this merged soon.
